test/server/cards/03_TWI/leaders/AsajjVentressUnparalleledAdversary.spec.ts

```diff
@@ -0,0 +1,114 @@
+describe('Asajj Ventress, Unparalleled Adversary', function () {
+    integration(function (contextRef) {
+        describe('Asajj Ventress\'s leader ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['smugglers-aid'],
+                        spaceArena: ['green-squadron-awing'],
+                        groundArena: ['battlefield-marine'],
+                        leader: 'asajj-ventress#unparalleled-adversary',
+                        resources: 3,
+                    },
+                    player2: {
+                        groundArena: ['admiral-yularen#advising-caution'],
+                    },
+                });
+            });
+
+            it('should initiate attack but does not give +1/+0 as we do not play any event this phase', function () {
+                const { context } = contextRef;
+
+                // initiate attack
+                context.player1.clickCard(context.asajjVentress);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.greenSquadronAwing]);
+                context.player1.clickCard(context.battlefieldMarine);
+                context.player1.clickCard(context.p2Base);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.asajjVentress.exhausted).toBeTrue();
+                expect(context.p2Base.damage).toBe(3);
+            });
+
+            it('should initiate attack and give +1/+0 as we play an event this phase', function () {
```

Comment:
Can we merge these tests with a `moveToNextActionPhase`, which will also confirm that it doesn't work for an event played in the previous phase